managers/views.py

```diff
@@ -202,3 +206,50 @@ def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
         task_activity.is_active = False
         task_activity.save()
         return HttpResponseRedirect(reverse("task-activities"))
+
+
+@method_decorator(login_required, name="dispatch")
+@method_decorator(
+    check_permissions(allowed_user_types=[CustomUser.UserType.ADMIN.name, CustomUser.UserType.MANAGER.name]),
+    name="dispatch",
+)
+class ProjectExpectedHoursView(DetailView):
+    model = Project
+    template_name = "managers/project_expected_hours.html"
+
+    def get_context_data(self, **kwargs: Any) -> dict:
+        context = super().get_context_data(**kwargs)
+        context["UI_text"] = ProjectExpectedHoursViewStrings
+        return context
+
+    def get_queryset(self) -> QuerySet:
+        return (
+            super()
+            .get_queryset()
+            .get_with_prefetched_expected_hours(
+                ExpectedHours.objects.get_expected_hours_for_a_particular_month(
+                    self.kwargs["year"], self.kwargs["month"]
+                )
+            )
+        )
+
+
+@method_decorator(login_required, name="dispatch")
+@method_decorator(
+    check_permissions(allowed_user_types=[CustomUser.UserType.ADMIN.name, CustomUser.UserType.MANAGER.name]),
+    name="dispatch",
+)
+class CreateExpectedHoursView(BSModalCreateView):
```

Comment:
I can't see anywhere if only assign to project manager can add expected hours for specific project. Please add test for that. If there is not such mechanism then you need to create decorator which should check whether manager is in project or not.

Reply:
Added `UserIsManagerOfCurrentProjectMixin` to `ProjectExpectedHoursView` and  `CreateExpectedHoursView`. Only manager of current project is able to see expected hours in the project as well as define new expected hours.